src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -5820,6 +5820,462 @@ public Derived() : base() { }
         );
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_01()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                [SetsRequiredMembers]
+                public Base(string str)
+                {
+                    this.Str = str;
+                }
+    
+                public required abstract string Str { get; set; }
+            }
+
+            public class Derived : Base
+            {
+                [SetsRequiredMembers]
+                public Derived(string str) : base(str)
+                {
+                }
+
+                public override required string Str { get; set; }
+            }
+            """;
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_02()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                [SetsRequiredMembers]
+                public Base(string str)
+                {
+                }
+
+                public required virtual string Str { get; set; } = "";
```

Comment:
So from this result, is the language effectively aware that a base constructor can satisfy its `required virtual` properties using property initializers, and skip calling the override accessors? But it can't do the same for a `required abstract`?

Reply:
No, that's pretty much the opposite conclusion 😆. The intuition is to think about the fields: for an `abstract` property, we know that the abstract definition cannot introduce a property and `base.Str` is illegal. For virtual properties, however, an override introduces a new slot and `base.Str` could produce a `null` where `Str` wouldn't. Look at `SetsRequiredMembersHonoredForPropertyOverride_06`; the `DerivedDevived` constructor has 2 warnings on it. One is for the current type implementation of `Str`, and one is for the non-`abstract` overridden `base.Str` property.

Reply:
> the DerivedDevived constructor has 2 warnings on it.

I think I forgot to ask, how should user resolve both warnings from the context of that constructor?

Reply:
> I guess constructor needs to assign both Str and base.Str separately?

Correct. I don't _think_ there's a slot they can't assign outside of existing holes, such as explicit interface implementation.